src/library_wasi.js

```diff
@@ -5,10 +5,66 @@
  * found in the LICENSE file.
  */
 
-mergeInto(LibraryManager.library, {
+var WasiLibrary = {
   proc_exit__deps: ['exit'],
   proc_exit: function(code) {
     return _exit(code);
   },
-});
 
+  emscripten_get_environ: function() {
+    if (!_emscripten_get_environ.ENV) {
+      var ENV = {};
+      ENV['USER'] = ENV['LOGNAME'] = 'web_user';
+      ENV['PATH'] = '/';
+      ENV['PWD'] = '/';
+      ENV['HOME'] = '/home/web_user';
+      // Browser language detection #8751
+      ENV['LANG'] = ((typeof navigator === 'object' &&navigator.languages && navigator.languages[0]) || 'C').replace('-', '_') + '.UTF-8';
+      ENV['_'] = thisProgram;
+      var strings = [];
+      for (var key in ENV) {
+        strings.push(key + '=' + ENV[key]);
+      }
+      _emscripten_get_environ.strings = strings;
+    }
```

Comment:
Shouldn't be here a setter for `_emscripten_get_environ.ENV` ? @kripken

Reply:
emscripten_get_environ is an internal helper function only.  I was just talking to @kripken about how we can internalize these kind of things so they are no exposed in any way.

As for a setter, its not possible for a program to set variables in the calling/parent environment.  This is, for example,  why wasi doesn't have a setenv syscall.   `setenv` and friends are all implemented internally within musl/libc.

Reply:
Thanks @syrusakbary! Yeah, there is a typo in this file - they should all be `_emscripten_get_environ.strings`. So this was doing a little extra work as the caching failed. Fix in #9529

Reply:
Oh sorry, looks like I the your comment.   Apologies.